server/mvcc/kvstore_txn.go

```diff
@@ -104,7 +104,6 @@ func (tw *storeTxnWrite) Put(key, value []byte, lease lease.LeaseID) int64 {
 func (tw *storeTxnWrite) End() {
 	// only update index if the txn modifies the mvcc state.
 	if len(tw.changes) != 0 {
-		tw.s.saveIndex(tw.tx)
```

Comment:
It looks like this breaks the atomicity of consistent index and writing transactions. @ptabor @ahrtr @serathius #13844